packages/callout-block/src/CalloutBlock.tsx

```diff
@@ -35,23 +59,28 @@ export const CalloutBlock = ({ appBridge }: BlockProps): ReactElement => {
         blockSettings.width === Width.HugContents && alignmentMap[blockSettings.alignment],
     ]);
 
-    const getAccentColor = (type: Type): string => {
-        const style = getComputedStyle(document.body);
-        switch (type) {
-            case Type.Info:
-                return style.getPropertyValue(`${THEME_PREFIX}accent-color-info-color`);
-            case Type.Note:
-                return style.getPropertyValue(`${THEME_PREFIX}accent-color-note-color`);
-            case Type.Tip:
-                return style.getPropertyValue(`${THEME_PREFIX}accent-color-tip-color`);
-            case Type.Warning:
-                return style.getPropertyValue(`${THEME_PREFIX}accent-color-warning-color`);
+    useEffect(() => {
+        const updateStyles = () => {
+            const { bgColor, txtColor } = computeStyles(type, appearance);
+            setBackgroundColor(bgColor);
+            setTextColor(txtColor);
+        };
+
+        const styleElement = document.getElementById('design-settings');
+        if (!styleElement) {
+            return;
         }
-    };
 
-    const accentColor = getAccentColor(blockSettings.type);
-    const backgroundColor = blockSettings.appearance === Appearance.Strong ? accentColor : setAlpha(0.1, accentColor);
-    const textColor = getTextColor(blockSettings.appearance, accentColor, backgroundColor);
+        const observer = new MutationObserver(() => {
```

Comment:
Instead of observer could you use the CSS variable and add a local CSS variable?

```html
<div id="upper" style="--my-var: blue;">
  <div id="upper" style="--my-var: red;"></div> <!-- become red -->
</div>
```

Reply:
Could do that, but the observer would still be needed, as the local variable would need to get recalculated after the change within the global theme settings.

Reply:
Optimally you shouldn't need the local variable, why is it needed?

Reply:
Because based on the setting `appearance` (Light/Dark), the background color gets calculated and also the text color. Which is based on the type (Info/Note/Warning/Tip), which then is the accent color that should be taken as base for the calculation.

Reply:
Instead of observing the DOM, could you listen to the `HubAppearanceUpdated` event that is emitted with window.emitter?

Reply:
That's most likely the solution, thanks :)